### PR TITLE
Tagging update

### PR DESCRIPTION
**Pull request by KiZach**
# Overview/Summary

Tagging support on resource group creation and logging resources

## This PR fixes/adds/changes/removes

1.	Add possibility to add tags to resource group module
2.	Add possibility to add tags to logging resources created by the logging module

### Breaking Changes

None

## Testing Evidence

Tested in 2 different environments with Azure Devops pipeline configured as below:

  - task: AzureCLI@2
    displayName: Az CLI Deploy Logging Resource Group
    name: create_logging_rg
    inputs:
      azureSubscription: $(ServiceConnectionName)
      scriptType: 'bash'
      scriptLocation: 'inlineScript'
      inlineScript: |
        az account set --subscription $(LoggingSubId)
        echo $(Tags)
        az deployment sub create \
        --template-file infra-as-code/bicep/modules/resourceGroup/resourceGroup.bicep \
        --parameters parResourceGroupName=$(LoggingResourceGroupName) parResourceGroupLocation=$(Location) parTags="$(Tags)" \
        --location $(Location) \
        --name create_logging_rg-$(RunNumber)

  - task: AzureCLI@2
    displayName: Az CLI Deploy Logging
    name: create_logging
    inputs:
      azureSubscription: $(ServiceConnectionName)
      scriptType: 'bash'
      scriptLocation: 'inlineScript'
      inlineScript: |
        az account set --subscription $(LoggingSubId)
        az deployment group create \
        --resource-group $(LoggingResourceGroupName) \
        --template-file infra-as-code/bicep/modules/logging/logging.bicep \
        --parameters parLogAnalyticsWorkspaceName=$(LogAnalyticsWorkspaceName) parLogAnalyticsWorkspaceLogRetentionInDays=$(LogAnalyticsRetentionInDays) parAutomationAccountName=$(AutomationAccountName) parTags="$(Tags)" \
        --name create_logging-$(RunNumber)

image

## As part of this Pull Request I have

- [X] Checked for duplicate [Pull Requests](https://github.com/Azure/ALZ-Bicep/pulls)
- [ ] Associated it with relevant [ADO items](https://dev.azure.com/unifiedactiontracker/Solution%20Engineering/_backlogs/backlog/Azure%20Landing%20Zone%20Bicep/Backlog%20Bugs%20Feedback)
- [X] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/Azure/ALZ-Bicep/tree/main)
- [X] Performed testing and provided evidence.
- [X] Updated relevant and associated documentation.

### CODE REVIEW

**Review by jtracey93 — CHANGES_REQUESTED** [review]
Hey @KiZach, thanks for the PR. Just one small thing to fix.

File: infra-as-code/bicep/modules/logging/README.md [hunk 1]
```diff
@@ -31,6 +31,7 @@ The module requires the following required input parameters.
...
 | parAutomationAccountName                   | string          | Automation account name                                     | Mandatory input, name must be unique in the subscription, default: `alz-automation-account`                                                                                                               | `alz-automation-account`                                                                                     |
 | parAutomationAccountRegion                 | string          | Region name                                                 | Mandatory input, default: `resourceGroup().location`                                                                                                                                                      | `eastus`                                                                                                     |
+| parTags                      | object | Empty object `{}`          | Array of Tags to be applied to all resources in the Spoke Network   | None        | `{"key": "value"}`                                                                                                                                    |
```
[the remark this answers is not in the document]
Comment (KiZach) [reply infra-as-code/bicep/modules/logging/README.md]: Both README.md files are now updated, and also above suggestion is implemented.

**Review by jtracey93 — APPROVED** [review]

**Comment by jtracey93** [comment]
Looking good, kicking off tests before merging

**Comment by jtracey93** [comment]
Only cleanup task failed which we will pickup offline in another issue. However, the deployment and other tests succeeded

image

**Pull request merged**